code/datums/disease.dm

```diff
@@ -78,27 +78,31 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 	return
 
 /datum/disease/proc/has_cure()//check if affected_mob has required reagents.
-	if(!cure_id) return 0
+	if((!cure_id && !cure_list) || !curable) return 0
 	var/result = 1
-	if(cure_list == list(cure_id))
-		if(istype(cure_id, /list))
+	if(!cure_list || cure_list == cure_id) //if no cure_list or cure list is the same as cure_id, just check _id
+		if(istype(cure_id, /list)) //if cure_id is a list, check inside of it
 			for(var/C_id in cure_id)
-				if(!affected_mob.reagents.has_reagent(C_id))
```

Comment:
Wait nevermind, I added it for a reason.

This is the list where it's checking all of the reagents in that list and checking if ONE of them is in you. If it fails the first reagent but passes on the second, then it's going to have result = 0 without the result = 1.